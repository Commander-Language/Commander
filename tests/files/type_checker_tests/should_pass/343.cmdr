//Test builtin substring type checks
a = 1;
x = "hi".substring(1);
x = "hi".substring(1, 2);
x = (1.0, true, 3)[a].substring(1);
x = (1.0, true, 3)[a].substring(1, 2);
x = "hi".substring((1.0, true, 3)[a]);
x = "hi".substring((1.0, true, 3)[a], (1.0, true, 3)[a]);
