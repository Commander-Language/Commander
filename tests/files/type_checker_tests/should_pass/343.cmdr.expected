(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode a) SET (IntExprNode 1 INT) INT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) substring (ExprsNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) (IntExprNode 1 INT)) STRING) STRING))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) substring (ExprsNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) (IntExprNode 1 INT) (IntExprNode 2 INT)) STRING) STRING))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) substring (ExprsNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) (IntExprNode 1 INT)) STRING) STRING))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) substring (ExprsNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) (IntExprNode 1 INT) (IntExprNode 2 INT)) STRING) STRING))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) substring (ExprsNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT))) STRING) STRING))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) substring (ExprsNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'hi') STRING))) STRING) (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT))) STRING) STRING)))
